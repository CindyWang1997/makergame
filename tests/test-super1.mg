object son : parent {
  event step {
    std::printstr("child step");
    super();
  }

  event draw {
    std::printstr("child draw");
    super();
  }
}

object daughter : parent {
  event step {
    super();
    std::printstr("daughter step");
  }

  event draw {
    super();
    std::printstr("daughter draw");
  }
}

object parent {
  event step { std::printstr("parent step"); }
  event draw { std::printstr("parent draw"); }
}

object main {
  int i;
  event create {
    i = 0;
    create son;
    create daughter;
  }

  event step { std::end_game(); }
}
