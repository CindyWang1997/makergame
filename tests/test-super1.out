child step
parent step
parent step
daughter step
child draw
parent draw
parent draw
daughter draw
